overlord/snapstate/snapstate.go

```diff
@@ -2054,3 +2070,36 @@ func GadgetConnections(st *state.State) ([]snap.GadgetConnection, error) {
 
 	return gadgetInfo.Connections, nil
 }
+
+// hook setup by devicestate
+var (
+	Model func(st *state.State) (*asserts.Model, error)
+)
+
+// ModelPastSeed returns the device model assertion if available and
+// the device is seeded, at that point the device store is known
+// and seeding done. Otherwise it returns a ChangeConflictError
+// about being too early.
+func ModelPastSeed(st *state.State) (*asserts.Model, error) {
```

Comment:
(nitpick^2) I wonder if we can find a more descriptive name. Some ideas: `ModelAfterSeeding`, `SeededModel`?